src/Papyros.ts

```diff
@@ -117,44 +58,60 @@ class PapyrosStateManager {
             this.statusPanel.setStatus(message || t(`Papyros.states.${state}`));
         }
     }
+
+    render(options: RenderOptions): HTMLElement {
+        return this.statusPanel.render(options);
+    }
 }
 
+const LANGUAGE_MAP = new Map([
+    ["python", ProgrammingLanguage.Python],
+    ["javascript", ProgrammingLanguage.JavaScript]
+]);
+
 interface PapyrosCodeState {
     programmingLanguage: ProgrammingLanguage;
     editor: CodeEditor;
     backend: Remote<Backend>;
     runId: number;
-    outputArea: HTMLInputElement;
 }
 
 interface PapyrosConfig {
     standAlone: boolean;
     programmingLanguage: ProgrammingLanguage;
     locale: string;
-    inputMode: InputMode
+    inputMode: InputMode;
+}
+
+interface PapyrosRenderOptions {
+    papyros: RenderOptions;
+    code?: RenderOptions;
+    panel?: RenderOptions;
+    input?: RenderOptions;
+    output?: RenderOptions
 }
 
 export class Papyros {
     stateManager: PapyrosStateManager;
     codeState: PapyrosCodeState;
     inputManager: InputManager;
     outputManager: OutputManager;
+    inputURL: string;
 
     constructor(programmingLanguage: ProgrammingLanguage, inputMode: InputMode) {
-        this.outputManager = new OutputManager(OUTPUT_TA_ID);
-        const statusPanel = new StatusPanel();
+        this.outputManager = new OutputManager();
         this.codeState = {
             programmingLanguage: programmingLanguage,
             editor: new CodeEditor(
-                document.getElementById(EDITOR_WRAPPER_ID) as HTMLInputElement,
-                programmingLanguage, statusPanel.render(document.createElement("div")),
+                programmingLanguage,
                 t("Papyros.code_placeholder", { programmingLanguage })),
             backend: {} as Remote<Backend>,
-            outputArea: document.getElementById(OUTPUT_TA_ID) as HTMLInputElement,
             runId: 0
         };
+        const statusPanel = new StatusPanel();
         this.stateManager = new PapyrosStateManager(statusPanel);
         this.inputManager = new InputManager(() => this.stateManager.setState(PapyrosState.Running), inputMode);
+        this.inputURL = location.host;
```

Comment:
Is this still used?

Reply:
The channel takes care of that now, so indeed no longer used.